internal/nhctl/controller/dev_start.go

```diff
@@ -603,7 +610,7 @@ func (c *Controller) genContainersAndVolumes(podSpec *corev1.PodSpec,
 		devImage = c.GetDevImage(containerName) // Default : replace the first container
 	}
 
-	sideCarContainer := generateSideCarContainer(c.GetDevSidecarImage(containerName), workDir)
+	sideCarContainer := generateSideCarContainer(c.GetDevSidecarImage(containerName), workDir, sshUsed)
```

Comment:
You cat get all `ContainerDevConfig` from `*Controller`.  
No need to add `genContainersAndVolumes()` input parameter `sshUsed bool`